shapeshift/web
# feat: multi-hop trade progression

## Description

- Upgrades lifi swapper to conditionally fetch multi-hop quotes (enabled for multi-hop only)
- Removes redux from thor swapper `getTradeQuote` to allow calling via RTK query
- Added thor swapper to multi-hop quote fetch
- Added debounce to multi-hop trades
- Implemented quote sorting based on input / output ratio

## Pull Request Type


## Issue (if applicable)

NA

## Risk

- High risk change to thor quotes and trades. The lift in removing redux from this swapper was fairly widespread so the testing should include multiple different chains (UTXO, EVM etc) and scenarios.
- Moderate risk to lifi quotes and trades. Logic has changed only slightly so just needs a sanity check that trades are still working.

## Testing

- Check thor quotes and trade execution
- Check lifi quotes and trade execution

### Engineering

See above

### Operations

See above

## Screenshots (if applicable)

Completed thor trade
https://etherscan.io/tx/0xfcad849d0d6764fcad5d99d8e9fec84f7f35e717813c882cc1c5ad686a920104
https://live.blockcypher.com/doge/tx/b0def029e603a18a9ea643c92c7e64e8a5854720210ebeacdf73f2a1b3cdb24a/
![image](https://github.com/shapeshift/web/assets/125113430/90bd857e-e220-48e4-8513-66a5fe4a6f72)

Completed lifi trade
https://optimistic.etherscan.io/tx/0x2d1a07967be9c009911a9265d831a349b1ae82d0ad211c0bcee507c14e810d71
https://snowtrace.io/tx/0x635ed06b909c0672b7162d031b9bfd6dbc4cf384e4b097db9f96408216eba655
![image](https://github.com/shapeshift/web/assets/125113430/c6d499de-9531-4bb1-b33b-7937a30f9f6d)
